File: payment-server/src/main/java/com/example/payment/dto/PaymentDto.java
package com.example.payment.dto;

import com.example.payment.entity.Payment;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PaymentDto {
    private Long id;
    private Long orderId;
    private Long userId;
    private BigDecimal amount;
    private Payment.PaymentStatus status;
    private Payment.PaymentMethod paymentMethod;
    private String transactionId;
    private LocalDateTime createdAt;
    private LocalDateTime paidAt;
}

File: payment-server/src/main/kotlin/com/example/payment/dto/PaymentDto.kt
package com.example.payment.dto

import com.example.payment.entity.Payment
import java.math.BigDecimal
import java.time.LocalDateTime

data class PaymentDto(
    val id: Long? = null,
    val orderId: Long,
    val userId: Long,
    val amount: BigDecimal,
    val status: Payment.PaymentStatus? = null,
    val paymentMethod: Payment.PaymentMethod? = null,
    val transactionId: String? = null,
    val createdAt: LocalDateTime? = null,
    val paidAt: LocalDateTime? = null
)
